pkg/fieldpath/fieldpath_test.go

```diff
@@ -212,6 +217,17 @@ func TestParse(t *testing.T) {
 				},
 			},
 		},
+		"QuotedFieldWithPeriod": {
+			reason: "A field name specified using quote can include a period",
+			path:   "metadata.labels[app.hash]",
```

Comment:
Did you mean for this test case to have `' '` around some fields? Otherwise, being quoted in this context just means the `path` variable is a string I believe, so not sure if it is testing the use case you want.

Reply:
Sorry, I have mistakes in writing case. I found this fieldPath package written by Nic is really good to use! I write this case to help users to know more use cases its support.